docs/modules/string.md

```diff
@@ -0,0 +1,424 @@
+# 🔠 String
+
+**`#bs.string:help`**
+
+Manage strings (text), allowing easy transformation and searching within the text.
+
+---
+
+## 🔧 Functions
+
+You can find below all functions available in this module.
+
+---
+
+### Concat
+
+```{function} #bs.string:concat
+
+Merge multiple strings from the input list into a single string.
+
+:Inputs:
+  **Storage `bs:in string.concat.list`**: {nbt}`string` List of strings to merge
+
+:Outputs:
+  **Storage `bs:out string.concat`**: {nbt}`string` The merged string
+```
+
+_Merge `Hello ` and `World`:_
+
+```mcfunction
+data modify storage bs:in string.concat.list set value ["Hello ","World"]
+function #bs.string:concat
+tellraw @a [{"text":"The merged string is \""},{"storage":"bs:out","nbt":"string.concat"},{"text":"\""}]
+```
+
+```{admonition} Technical Limitation
+:class: warning
+
+Due to the way string are handled, we don't support using `"` you need to use the `\"` instead.
+
+```
+
+> **Credits**: Aure31
+
+---
+
+### Convert Case
+
+:::::{tab-set}
+::::{tab-item} Uppercase
+
+```{function} #bs.string:upper
+
+Convert text to uppercase.
+
+:Inputs:
+  **Storage `bs:in string.upper.str`**: {nbt}`string` The string to convert to uppercase.
+
+:Outputs:
+  **Storage `bs:out string.upper`**: {nbt}`string` The uppercase string.
+```
+
+_Convert `hello world` to its uppercase version `HELLO WORLD`:_
+
+```mcfunction
+data modify storage bs:in string.upper.str set value "hello world"
+function #bs.string:upper
+tellraw @a [{"text":"The uppercase string is \""},{"storage":"bs:out","nbt":"string.upper"},{"text":"\""}]
+```
+
+::::
+::::{tab-item} Lowercase
+
+```{function} #bs.string:lower
+
+Convert text to lowercase.
+
+:Inputs:
+  **Storage `bs:in string.lower.str`**: {nbt}`string` The string to convert to lowercase.
+
+:Outputs:
+  **Storage `bs:out string.lower`**: {nbt}`string` The lowercase string.
+```
+
+_Convert `HELLO WORLD` to its lowercase version `hello world`:_
+
+```mcfunction
+data modify storage bs:in string.lower.str set value "HELLO WORLD"
+function #bs.string:lower
+tellraw @a [{"text":"The lowercase string is \""},{"storage":"bs:out","nbt":"string.lower"},{"text":"\""}]
+```
+
+::::
+:::::
+
+```{admonition} Support
+:class: hint
+
+this function actually support all UTF-8 characters so you can use it with any language.
```

Comment:
"This function actually supports all UTF-8 characters."
I don't think we need to specify we "can use it with any language".